tests/foreman/cli/test_syncplan.py

```diff
@@ -273,3 +319,271 @@ def test_verify_bugzilla_1261122(self):
         new_sync_plan = self._make_sync_plan()
         result = SyncPlan.info({'id': new_sync_plan['id']})
         self.assertIsNotNone(result.get('enabled'))
+
+    # This Bugzilla bug is private. It is impossible to fetch info about it.
+    @stubbed('Unstab when BZ1279539 is fixed')
+    @tier1
+    def test_verify_bugzilla_1279539(self):
+        """Verify product won't get synced immediately after adding association
+        with a sync plan which has already been started
+
+        @Feature: Sync Plan
+
+        @Assert: Repository was not synchronized
+
+        @BZ: 1279539
+        """
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': self.org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        product = make_product({'organization-id': self.org['id']})
+        repo = make_repository({'product-id': product['id']})
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        with self.assertRaises(AssertionError):
+            self.validate_repo_content(
+                repo, ['errata', 'package-groups', 'packages'])
+
+    # This Bugzilla bug is private. It is impossible to fetch info about it.
+    @stubbed('Unstab when BZ1279539 is fixed')
+    @tier4
+    def test_positive_synchronize_custom_product_current_sync_date(self):
+        """Create a sync plan with current datetime as a sync date, add a
+        custom product and verify the product gets synchronized on the next
+        sync occurrence
+
+        @Assert: Product is synchronized successfully.
+
+        @Feature: SyncPlan
+
+        @BZ: 1279539
+        """
+        interval = 60 * 60  # 'hourly' sync interval in seconds
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'interval': 'hourly',
+            'organization-id': self.org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
```

Comment:
how long are we waiting in this test for the next sync to run?

Just to let you know you can set the sync-date to future time - probably 30 sec or 1 min from current time so you dont need to wait longer for the next sync.